repair/row_level.cc

```diff
@@ -342,18 +342,24 @@ class repair_row {
     std::optional<repair_sync_boundary> _boundary;
     std::optional<repair_hash> _hash;
     lw_shared_ptr<mutation_fragment> _mf;
+    bool _flush_to_disk;
 public:
     repair_row() = default;
     repair_row(std::optional<frozen_mutation_fragment> fm,
             std::optional<position_in_partition> pos,
             lw_shared_ptr<const decorated_key_with_hash> dk_with_hash,
             std::optional<repair_hash> hash,
-            lw_shared_ptr<mutation_fragment> mf = {})
+            lw_shared_ptr<mutation_fragment> mf = {},
+            bool flush_to_disk = false)
             : _fm(std::move(fm))
             , _dk_with_hash(std::move(dk_with_hash))
             , _boundary(pos ? std::optional<repair_sync_boundary>(repair_sync_boundary{_dk_with_hash->dk, std::move(*pos)}) : std::nullopt)
             , _hash(std::move(hash))
-            , _mf(std::move(mf)) {
+            , _mf(std::move(mf))
+            , _flush_to_disk(flush_to_disk) {
+    }
```

Comment:
I don't understand this flush_to_disk. Is it needed to distinguish between rows that are unequal on the repair master before the operation and after the operation, and so need to be written?

Reply:
It is used to distinguish if the row is recieved from peers. Repair master only receive rows from peers that the master does not have. So if a row is recieved from master, it needs to write to disk.

Reply:
Ok. Suggestions for readability:
 - use bool_class so it's clear on the caller side
 - rename to dirty_on_master or dirty_locally, this informs us about the row, not what you want to do with it. `flush_to_disk` doesn't tell us why
 - undefault the parameter, defaults are often dangerous, and here there is no natural reason for a row to be not-dirty

Reply:
Ok. Done. Fixed in latest version.